test_dependencies.md

```diff
@@ -1,21 +1,75 @@
-Known dependencies for testing kb_sdk itself (not modules) listed below. Please add more if discovered.
+# Test Dependencies
 
-Must force RPC::Any, Moose updates cause XML tests to fail.
+All of the following commands were used to set up the VirtualBox/Vagrant image that is used for tests and development. You don't need to do any of this on your own machine if you run tests through Vagrant. This is only kept here as a record of all the required steps in case we ever need to rebuild our image.
 
-New versions of paramiko complain about being unable to import SSH2_AGENTC_REQUEST_IDENTITIES, 1.10.2 works.
+## Basics
 
-With the new auth2 clients and server we may no longer need rsa, paramiko, Crypt::OpenSSL::RSA, Convert::PEM
+```sh
+$ sudo apt-get update
+$ sudo apt-get install build-essential libfontconfig
+```
+
+## Java
 
-Make sure nose2 is using python2.
+```sh
+$ sudo apt-get remove openjdk*
+$ sudo apt-get autoclean
+$ sudo apt-get install openjdk-8-jdk ant
+```
 
-pip install requests nose2 jsonrpcbase rsa paramiko==1.10.2
+## Docker
 
-cpan Moose
+```sh
+$ sudo apt-get update
+$ sudo apt-get install apt-transport-https ca-certificates curl software-properties-common
+$ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+$ sudo apt-key fingerprint 0EBFCD88
+$ sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+$ sudo apt-get update
+$ sudo apt-get install docker-ce
+```
 
-cpan -f -i RPC::Any
+## JS
 
-cpan Bio::KBase::Exceptions
+CasperJS, > 1.1.0. http://casperjs.org/ 
 
-cpan Parse::Yapp Devel::StackTrace Lingua::EN::Inflect Template File::Slurp Cwd JSON Data::UUID XML::Dumper JSON::RPC::Client Exception::Class Config::Simple Digest::SHA1 Crypt::OpenSSL::RSA Convert::PEM DateTime MIME::Base64 URI Object::Tiny::RW Plack File::ShareDir::Install YAML TAP::Harness Plack::Middleware::CrossOrigin RPC::Any::Server::JSONRPC::PSGI
+```sh
+$ git clone git://github.com/casperjs/casperjs.git`
+$ cd casperjs
+$ ln -sf `pwd`/bin/casperjs /usr/local/bin/casperjs
+```
 
-CasperJS must be installed, > 1.1.0. http://casperjs.org/ .  PhantomJS 1.9.x also required
+PhantomJS 1.9.x
+```sh
+$ export PHANTOM_JS="phantomjs-1.9.8-linux-x86_64"
+$ wget https://bitbucket.org/ariya/phantomjs/downloads/$PHANTOM_JS.tar.bz2
+$ tar xvjf $PHANTOM_JS.tar.bz2
+$ mv $PHANTOM_JS /opt/
+$ ln -sf /opt/$PHANTOM_JS/bin/phantomjs /usr/local/bin
+```
+
+## Python
+
+```sh
+sudo apt-get install python python-setuptools
+easy_install pip
+pip install requests nose2 jsonrpcbase
+```
+
+New versions of paramiko complain about being unable to import SSH2_AGENTC_REQUEST_IDENTITIES, 1.10.2 works.
```

Comment:
Looks like the paramiko install has been removed, but this line is still here. Do we need paramiko with the new clients?